src/solver/solver.go

```diff
@@ -117,14 +128,12 @@ func (r *resolver) resolveTypeNoLateStaticBinding(class, typ string) map[string]
 	case meta.WInstanceMethodCall:
 		expr, methodName := meta.UnwrapInstanceMethodCall(typ)
 
-		for className := range r.resolveType(class, expr) {
-			info, _, ok := FindMethod(className, methodName)
-			if ok {
-				for tt := range r.resolveTypes(className, info.Typ) {
-					res[tt] = struct{}{}
-				}
-			}
+		instanceTypes := r.resolveType(class, expr)
+		r.collectMethodCallTypes(res, instanceTypes, methodName)
+		if len(res) == 0 {
```

Comment:
I am fairly certain that `res` is going to be always empty here as there is no code that updates it anymore.

Reply:
Oh, I now see that `collectMethodCallTypes` actually fills in res. I would suggest to not modify the result map this way, or, for example, return it back (and call the method like "appendMethodCallTypes" instead) so that is more easily noticable.

Reply:
I think my main reason to do that was a fact that we already allocate a map on a caller side, so it felt wasteful to allocate a new one and to re-assign that to a caller-side map.

I agree that this is not worthwhile. :)
I'll fix that to make code more easy to read.